transitions/core.py

```diff
@@ -290,7 +290,7 @@ def execute(self, event_data):
     def _change_state(self, event_data):
         event_data.machine.get_state(self.source).exit(event_data)
         event_data.machine.set_state(self.dest, event_data.model)
-        event_data.update(event_data.model.state)
+        event_data.update(event_data.machine.get_model_state(event_data.model).value)
```

Comment:
Yes it should suffice. I chose to use an explicit method for this so that other classes don't have to know the internal details on how to get model state. Removing that would make it something like this:

```python
event_data.update(getattr(event_data.model, event_data.machine.model_attribute))
```

Should I go ahead and change it in all other places too?

Reply:
> Should I go ahead and change it in all other places too?
yes, please.

Reply:
Fixed in 9163fed.